packages/mobile/src/identity/reducer.ts

```diff
@@ -349,7 +362,42 @@ export const secureSendPhoneNumberMappingSelector = (state: RootState) =>
 export const importContactsProgressSelector = (state: RootState) =>
   state.identity.importContactsProgress
 export const matchedContactsSelector = (state: RootState) => state.identity.matchedContacts
-export const verificationStateSelector = (state: RootState) => state.identity.verificationState
+
+export const isBalanceSufficientForSigRetrievalSelector = (state: RootState) => {
+  const dollarBalance = stableTokenBalanceSelector(state) || 0
+  const celoBalance = celoTokenBalanceSelector(state) || 0
+  return isBalanceSufficientForSigRetrieval(dollarBalance, celoBalance)
+}
+
+function isBalanceSufficientForAttestations(state: RootState, attestationsRemaining: number) {
+  const userBalance = stableTokenBalanceSelector(state) || 0
+  return new BigNumber(userBalance).isGreaterThan(
+    attestationsRemaining * ESTIMATED_COST_PER_ATTESTATION
+  )
+}
+
+const identityVerificationStateSelector = (state: RootState) => state.identity.verificationState
+
+const isBalanceSufficientSelector = (state: RootState) => {
+  const verificationState = state.identity.verificationState
+  const { phoneHashDetails, status, actionableAttestations } = verificationState
+  const attestationsRemaining = status.numAttestationsRemaining - actionableAttestations.length
+  const isBalanceSufficient = !phoneHashDetails.phoneHash
```

Comment:
If:
* we do not have `phoneHash`
* we do have enough money to retrieve it
* we do not have enough money for attestations

this would return `TRUE`, right?

Reply:
Yes, it doesn't seem right.

Hum problem is we should add the sum which is not directly possible with the current helpers.

Reply:
hum even summing amounts wouldn't be right
cause you can have just enough to fetch the pepper
but if you're already verified you don't need more for attestations (edited) 
it's really a 2 steps process

do you see a different way to solve this?

Reply:
Actually, the only case when the issue, I’ve pointed, would occur is when fetching pepper has returned an error. User then would be able to start the process even if it was not enough money for attestations. But then if we fetch the pepper eventually - we would show them that balance is insufficient for attestations as a popup anyway. I thinks this is ok for the edge case, when we can not estimate amount needed for attestations.